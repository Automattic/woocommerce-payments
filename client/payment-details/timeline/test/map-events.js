/** @format */
/**
 * Internal dependencies
 */
import mapTimelineEvents from '../map-events';

describe( 'mapTimelineEvents', () => {
	beforeEach( () => {
		jest.clearAllMocks();
		global.wcpaySettings = {
			featureFlags: { capital: false },
			zeroDecimalCurrencies: [],
		};
	} );

	test( 'handles falsey values', () => {
		expect( mapTimelineEvents( null ) ).toStrictEqual( [] );
	} );

	test( 'formats authorized events', () => {
		expect(
			mapTimelineEvents( [
				{
					amount: 7900,
					currency: 'USD',
					datetime: 1585589596,
					type: 'authorized',
				},
			] )
		).toMatchSnapshot();
	} );

	test( 'formats authorization_voided events', () => {
		expect(
			mapTimelineEvents( [
				{
					amount: 5900,
					currency: 'USD',
					datetime: 1585652279,
					type: 'authorization_voided',
				},
			] )
		).toMatchSnapshot();
	} );

	test( 'formats authorization_expired events', () => {
		expect(
			mapTimelineEvents( [
				{
					amount: 8600,
					currency: 'USD',
					datetime: 1585691920,
					type: 'authorization_expired',
				},
			] )
		).toMatchSnapshot();
	} );

	test( 'formats card_declined events', () => {
		expect(
			mapTimelineEvents( [
				{
					amount: 7700,
					currency: 'USD',
					datetime: 1585712113,
					reason: 'card_declined',
					type: 'failed',
				},
			] )
		).toMatchSnapshot();
	} );

	test( 'formats dispute_needs_response events with no amount', () => {
		expect(
			mapTimelineEvents( [
				{
					amount: null,
					currency: null,
					datetime: 1585793174,
					deposit: null,
					dispute_id: 'some_id',
					evidence_due_by: 1585879574,
					fee: null,
					reason: 'fraudulent',
					type: 'dispute_needs_response',
				},
			] )
		).toMatchSnapshot();
	} );

	test( 'formats dispute_warning_closed events', () => {
		expect(
			mapTimelineEvents( [
				{
					datetime: 1585793174,
					type: 'dispute_warning_closed',
				},
			] )
		).toMatchSnapshot();
	} );

	test( 'formats dispute_charge_refunded events', () => {
		expect(
			mapTimelineEvents( [
				{
					datetime: 1585793174,
					type: 'dispute_charge_refunded',
				},
			] )
		).toMatchSnapshot();
	} );

	test( 'formats dispute_in_review events', () => {
		expect(
			mapTimelineEvents( [
				{
					datetime: 1585859207,
					type: 'dispute_in_review',
					user_id: 1,
				},
			] )
		).toMatchSnapshot();
	} );

	describe( 'single currency events', () => {
		test( 'formats captured events without fee details', () => {
			expect(
				mapTimelineEvents( [
					{
						amount: 6300,
						currency: 'USD',
						datetime: 1585751874,
						deposit: {
							arrival_date: 1585838274,
							id: 'dummy_po_5eaada696b281',
						},
						fee: 350,
						type: 'captured',
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'should not render fee breakup when fee history is not present', () => {
			expect(
				mapTimelineEvents( [
					{
						amount: 6300,
						currency: 'USD',
						datetime: 1585751874,
						deposit: {
							arrival_date: 1585838274,
							id: 'dummy_po_5eaada696b281',
						},
						fee: 350,
						fee_rates: {
							percentage: 0.0195,
							fixed: 15,
							fixed_currency: 'USD',
						},
						type: 'captured',
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'formats captured events with fee details', () => {
			expect(
				mapTimelineEvents( [
					{
						amount: 6300,
						currency: 'USD',
						datetime: 1585751874,
						deposit: {
							arrival_date: 1585838274,
							id: 'dummy_po_5eaada696b281',
						},
						fee: 350,
						fee_rates: {
							percentage: 0.0195,
							fixed: 15,
							fixed_currency: 'USD',
							history: [
								{
									type: 'base',
									percentage_rate: 0.014,
									fixed_rate: 20,
									currency: 'gbp',
								},
								{
									type: 'additional',
									additional_type: 'international',
									percentage_rate: 0.014999999999999998,
									fixed_rate: 0,
									currency: 'gbp',
								},
								{
									type: 'additional',
									additional_type: 'fx',
									percentage_rate: 0.020000000000000004,
									fixed_rate: 0,
									currency: 'gbp',
								},
								{
									type: 'discount',
									percentage_rate: -0.049,
									fixed_rate: -20,
									currency: 'gbp',
								},
							],
						},
						type: 'captured',
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'formats captured events with just the base fee', () => {
			expect(
				mapTimelineEvents( [
					{
						amount: 6300,
						currency: 'USD',
						datetime: 1585751874,
						deposit: {
							arrival_date: 1585838274,
							id: 'dummy_po_5eaada696b281',
						},
						fee: 350,
						fee_rates: {
							percentage: 0.0195,
							fixed: 15,
							fixed_currency: 'USD',
							history: [
								{
									type: 'base',
									percentage_rate: 0.014,
									fixed_rate: 20,
									currency: 'gbp',
								},
							],
						},
						type: 'captured',
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'formats dispute_needs_response events', () => {
			expect(
				mapTimelineEvents( [
					{
						amount: 9500,
						currency: 'USD',
						datetime: 1585793174,
						deposit: null,
						dispute_id: 'some_id',
						evidence_due_by: 1585879574,
						fee: 1500,
						reason: 'fraudulent',
						type: 'dispute_needs_response',
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'formats partial_refund events', () => {
			expect(
				mapTimelineEvents( [
					{
						amount_refunded: 5000,
						currency: 'USD',
						datetime: 1585940281,
						deposit: null,
						type: 'partial_refund',
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'formats full_refund events', () => {
			expect(
				mapTimelineEvents( [
					{
						amount_refunded: 10000,
						currency: 'USD',
						datetime: 1586008266,
						deposit: null,
						type: 'full_refund',
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'formats dispute_won events', () => {
			expect(
				mapTimelineEvents( [
					{
						amount: 10000,
						currency: 'USD',
						datetime: 1586017250,
						deposit: {
							arrival_date: 1586103650,
							id: 'dummy_po_5eaada696b2d3',
						},
						fee: 1500,
						type: 'dispute_won',
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'formats dispute_lost events', () => {
			expect(
				mapTimelineEvents( [
					{
						amount: 10000,
						currency: 'USD',
						datetime: 1586055370,
						deposit: {
							arrival_date: 1586141770,
							id: 'dummy_po_5eaada696b2ef',
						},
						fee: 1500,
						type: 'dispute_lost',
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'formats financing paydown events', () => {
			global.wcpaySettings.featureFlags.capital = true;
			expect(
				mapTimelineEvents( [
					{
						type: 'financing_paydown',
						date: 1643717044,
						amount: -11000,
						loan_id: 'flxln_1KOKzdR4ByxURRrFX9A65q40',
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'hides financing paydown events when capital is disabled', () => {
			global.wcpaySettings.featureFlags.capital = false;

			expect(
				mapTimelineEvents( [
					{
						type: 'financing_paydown',
						date: 1643717044,
						amount: -11000,
						loan_id: 'flxln_1KOKzdR4ByxURRrFX9A65q40',
					},
				] )
			).toEqual( [] );
		} );
	} );

	describe( 'Multi-Currency events', () => {
		test( 'formats captured events without fee details', () => {
			expect(
				mapTimelineEvents( [
					{
						amount: 1800,
						currency: 'EUR',
						datetime: 1585751874,
						deposit: {
							arrival_date: 1585838274,
							id: 'dummy_po_5eaada696b281',
						},
						fee: 52,
						type: 'captured',
						transaction_details: {
							customer_amount: 1800,
							customer_currency: 'EUR',
							customer_fee: 52,
							store_amount: 2159,
							store_currency: 'USD',
							store_fee: 62,
						},
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'formats captured events with fee details', () => {
			expect(
				mapTimelineEvents( [
					{
						amount: 1800,
						currency: 'EUR',
						datetime: 1585751874,
						deposit: {
							arrival_date: 1585838274,
							id: 'dummy_po_5eaada696b281',
						},
						fee: 52,
						fee_rates: {
							percentage: 0.029,
							fixed: 30,
							fixed_currency: 'USD',
						},
						type: 'captured',
						transaction_details: {
							customer_amount: 1800,
							customer_currency: 'EUR',
							customer_fee: 52,
							store_amount: 2159,
							store_currency: 'USD',
							store_fee: 62,
						},
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'formats dispute_needs_response events', () => {
			expect(
				mapTimelineEvents( [
					{
						amount: -2160,
						currency: 'USD',
						datetime: 1585793174,
						deposit: null,
						dispute_id: 'some_id',
						evidence_due_by: 1585879574,
						fee: 1500,
						reason: 'fraudulent',
						type: 'dispute_needs_response',
						transaction_details: {
							customer_amount: 1800,
							customer_currency: 'EUR',
							customer_fee: null,
							store_amount: -2160,
							store_currency: 'USD',
							store_fee: 1500,
						},
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'formats partial_refund events', () => {
			expect(
				mapTimelineEvents( [
					{
						amount_refunded: 500,
						currency: 'EUR',
						datetime: 1585940281,
						deposit: null,
						type: 'partial_refund',
						transaction_details: {
							customer_amount: 500,
							customer_currency: 'EUR',
							customer_fee: 0,
							store_amount: 600,
							store_currency: 'USD',
							store_fee: 0,
						},
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'formats full_refund events', () => {
			expect(
				mapTimelineEvents( [
					{
						amount_refunded: 1800,
						currency: 'EUR',
						datetime: 1586008266,
						deposit: null,
						type: 'full_refund',
						transaction_details: {
							customer_amount: 1800,
							customer_currency: 'EUR',
							customer_fee: 0,
							store_amount: 2164,
							store_currency: 'USD',
							store_fee: 0,
						},
					},
				] )
			).toMatchSnapshot();
		} );

		test( 'formats dispute_won events', () => {
			expect(
				mapTimelineEvents( [
					{
						amount: 2999,
						currency: 'USD',
						datetime: 1586017250,
						deposit: {
							arrival_date: 1586103650,
							id: 'dummy_po_5eaada696b2d3',
						},
						fee: -1500,
						type: 'dispute_won',
						transaction_details: {
							customer_amount: 2500,
							customer_currency: 'EUR',
							customer_fee: null,
							store_amount: 2999,
							store_currency: 'USD',
							store_fee: -1500,
						},
					},
				] )
			).toMatchSnapshot();
		} );
	} );
} );
