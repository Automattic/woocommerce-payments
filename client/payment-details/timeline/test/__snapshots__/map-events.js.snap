// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapTimelineEvents Multi-Currency events formats captured events with fee details 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": "Payment status changed to Paid.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": <React.Fragment>
      $20.97 USD was added to your 
      <Link
        href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=dummy_po_5eaada696b281"
        type="wc-admin"
      >
        Apr 2, 2020 deposit
      </Link>
      .
    </React.Fragment>,
    "icon": <t
      className=""
      icon="plus"
      size={24}
    />,
  },
  Object {
    "body": Array [
      "1,00 EUR → 1.19944 USD: $21.59 USD",
      "Fee (2.9% + $0.30): $-0.62",
      undefined,
      "Net deposit: $20.97 USD",
    ],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": "A payment of €18,00 EUR was successfully charged.",
    "icon": <t
      className="is-success"
      icon="checkmark"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents Multi-Currency events formats captured events without fee details 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": "Payment status changed to Paid.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": <React.Fragment>
      $20.97 USD was added to your 
      <Link
        href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=dummy_po_5eaada696b281"
        type="wc-admin"
      >
        Apr 2, 2020 deposit
      </Link>
      .
    </React.Fragment>,
    "icon": <t
      className=""
      icon="plus"
      size={24}
    />,
  },
  Object {
    "body": Array [
      "1,00 EUR → 1.19944 USD: $21.59 USD",
      "Fee: €0,52",
      undefined,
      "Net deposit: $20.97 USD",
    ],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": "A payment of €18,00 EUR was successfully charged.",
    "icon": <t
      className="is-success"
      icon="checkmark"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents Multi-Currency events formats dispute_needs_response events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-02T02:06:14.000Z,
    "headline": "Payment status changed to Disputed: Needs response.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [
      "Disputed amount: €18,00",
      "1,00 EUR → 1.2 USD: $21.60 USD",
      "Fee: $15.00",
    ],
    "date": 2020-04-02T02:06:14.000Z,
    "headline": "$36.60 USD will be deducted from a future deposit.",
    "icon": <t
      className=""
      icon="minus"
      size={24}
    />,
  },
  Object {
    "body": Array [
      <Link
        data-testid="view-dispute-button"
        href="admin.php?page=wc-admin&path=%2Fpayments%2Fdisputes%2Fdetails&id=some_id"
        type="wc-admin"
      >
        View dispute
      </Link>,
    ],
    "date": 2020-04-02T02:06:14.000Z,
    "headline": "Payment disputed as Fraudulent.",
    "icon": <t
      className="is-error"
      icon="cross"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents Multi-Currency events formats dispute_won events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-04T16:20:50.000Z,
    "headline": "Payment status changed to Disputed: Won.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [
      "Dispute reversal: $29.99",
      "Fee refund: $15.00",
    ],
    "date": 2020-04-04T16:20:50.000Z,
    "headline": <React.Fragment>
      $44.99 USD was added to your 
      <Link
        href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=dummy_po_5eaada696b2d3"
        type="wc-admin"
      >
        Apr 5, 2020 deposit
      </Link>
      .
    </React.Fragment>,
    "icon": <t
      className=""
      icon="plus"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-04T16:20:50.000Z,
    "headline": "Dispute won! The bank ruled in your favor.",
    "icon": <t
      className="is-success"
      icon="notice-outline"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents Multi-Currency events formats full_refund events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-04T13:51:06.000Z,
    "headline": "Payment status changed to Refunded.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-04T13:51:06.000Z,
    "headline": "$21.64 USD will be deducted from a future deposit.",
    "icon": <t
      className=""
      icon="minus"
      size={24}
    />,
  },
  Object {
    "body": Array [
      "1,00 EUR → 1.20222 USD: $21.64 USD",
    ],
    "date": 2020-04-04T13:51:06.000Z,
    "headline": "A payment of €18,00 EUR was successfully refunded.",
    "icon": <t
      className="is-success"
      icon="checkmark"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents Multi-Currency events formats partial_refund events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-03T18:58:01.000Z,
    "headline": "Payment status changed to Partial refund.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-03T18:58:01.000Z,
    "headline": "$6.00 USD will be deducted from a future deposit.",
    "icon": <t
      className=""
      icon="minus"
      size={24}
    />,
  },
  Object {
    "body": Array [
      "1,00 EUR → 1.2 USD: $6.00 USD",
    ],
    "date": 2020-04-03T18:58:01.000Z,
    "headline": "A payment of €5,00 EUR was successfully refunded.",
    "icon": <t
      className="is-success"
      icon="checkmark"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents formats authorization_expired events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-03-31T21:58:40.000Z,
    "headline": "Payment status changed to Authorization expired.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-03-31T21:58:40.000Z,
    "headline": "Authorization for $86.00 USD expired.",
    "icon": <t
      className="is-error"
      icon="cross"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents formats authorization_voided events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-03-31T10:57:59.000Z,
    "headline": "Payment status changed to Authorization voided.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-03-31T10:57:59.000Z,
    "headline": "Authorization for $59.00 USD was voided.",
    "icon": <t
      className="is-warning"
      icon="checkmark"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents formats authorized events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-03-30T17:33:16.000Z,
    "headline": "Payment status changed to Authorized.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-03-30T17:33:16.000Z,
    "headline": "A payment of $79.00 USD was successfully authorized.",
    "icon": <t
      className="is-warning"
      icon="checkmark"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents formats card_declined events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-01T03:35:13.000Z,
    "headline": "Payment status changed to Failed.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-01T03:35:13.000Z,
    "headline": "A payment of $77.00 USD failed.",
    "icon": <t
      className="is-error"
      icon="cross"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents formats dispute_charge_refunded events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-02T02:06:14.000Z,
    "headline": "The disputed charge has been refunded.",
    "icon": <t
      className="is-success"
      icon="notice-outline"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents formats dispute_in_review events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-02T20:26:47.000Z,
    "headline": "Payment status changed to Disputed: In review.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-02T20:26:47.000Z,
    "headline": "Challenge evidence submitted.",
    "icon": <t
      className="is-success"
      icon="checkmark"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents formats dispute_needs_response events with no amount 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-02T02:06:14.000Z,
    "headline": "Payment status changed to Disputed: Needs response.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [
      "The cardholder's bank is requesting more information to decide whether to return these funds to the cardholder.",
    ],
    "date": 2020-04-02T02:06:14.000Z,
    "headline": "No funds have been withdrawn yet.",
    "icon": <t
      className=""
      icon="info-outline"
      size={24}
    />,
  },
  Object {
    "body": Array [
      <Link
        data-testid="view-dispute-button"
        href="admin.php?page=wc-admin&path=%2Fpayments%2Fdisputes%2Fdetails&id=some_id"
        type="wc-admin"
      >
        View dispute
      </Link>,
    ],
    "date": 2020-04-02T02:06:14.000Z,
    "headline": "Payment disputed as Fraudulent.",
    "icon": <t
      className="is-error"
      icon="cross"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents formats dispute_warning_closed events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-02T02:06:14.000Z,
    "headline": "Dispute inquiry closed. The bank chose not to pursue this dispute.",
    "icon": <t
      className="is-success"
      icon="notice-outline"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents single currency events formats captured events with fee details 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": "Payment status changed to Paid.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": <React.Fragment>
      $59.50 USD was added to your 
      <Link
        href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=dummy_po_5eaada696b281"
        type="wc-admin"
      >
        Apr 2, 2020 deposit
      </Link>
      .
    </React.Fragment>,
    "icon": <t
      className=""
      icon="plus"
      size={24}
    />,
  },
  Object {
    "body": Array [
      undefined,
      "Fee (1.95% + $0.15): $-3.50",
      <ul
        className="fee-breakdown-list"
      >
         
        <li>
          Base fee: 1.4% + £0.20
        </li>
        <li>
          International card fee: 1.5%
        </li>
        <li>
          Foreign exchange fee: 2%
        </li>
        <li>
          Discount
          <ul
            className="discount-split-list"
          >
            <li>
              Variable fee: 
              -4.9
              %
            </li>
            <li>
              Fixed fee: 
              £-0.20
            </li>
          </ul>
        </li>
         
      </ul>,
      "Net deposit: $59.50 USD",
    ],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": "A payment of $63.00 USD was successfully charged.",
    "icon": <t
      className="is-success"
      icon="checkmark"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents single currency events formats captured events with just the base fee 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": "Payment status changed to Paid.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": <React.Fragment>
      $59.50 USD was added to your 
      <Link
        href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=dummy_po_5eaada696b281"
        type="wc-admin"
      >
        Apr 2, 2020 deposit
      </Link>
      .
    </React.Fragment>,
    "icon": <t
      className=""
      icon="plus"
      size={24}
    />,
  },
  Object {
    "body": Array [
      undefined,
      "Base fee (1.95% + $0.15): $-3.50",
      undefined,
      "Net deposit: $59.50 USD",
    ],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": "A payment of $63.00 USD was successfully charged.",
    "icon": <t
      className="is-success"
      icon="checkmark"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents single currency events formats captured events without fee details 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": "Payment status changed to Paid.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": <React.Fragment>
      $59.50 USD was added to your 
      <Link
        href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=dummy_po_5eaada696b281"
        type="wc-admin"
      >
        Apr 2, 2020 deposit
      </Link>
      .
    </React.Fragment>,
    "icon": <t
      className=""
      icon="plus"
      size={24}
    />,
  },
  Object {
    "body": Array [
      undefined,
      "Fee: $3.50",
      undefined,
      "Net deposit: $59.50 USD",
    ],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": "A payment of $63.00 USD was successfully charged.",
    "icon": <t
      className="is-success"
      icon="checkmark"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents single currency events formats dispute_lost events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-05T02:56:10.000Z,
    "headline": "Payment status changed to Disputed: Lost.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-05T02:56:10.000Z,
    "headline": "Dispute lost. The bank ruled in favor of your customer.",
    "icon": <t
      className="is-error"
      icon="cross"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents single currency events formats dispute_needs_response events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-02T02:06:14.000Z,
    "headline": "Payment status changed to Disputed: Needs response.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [
      "Disputed amount: $95.00",
      undefined,
      "Fee: $15.00",
    ],
    "date": 2020-04-02T02:06:14.000Z,
    "headline": "$110.00 USD will be deducted from a future deposit.",
    "icon": <t
      className=""
      icon="minus"
      size={24}
    />,
  },
  Object {
    "body": Array [
      <Link
        data-testid="view-dispute-button"
        href="admin.php?page=wc-admin&path=%2Fpayments%2Fdisputes%2Fdetails&id=some_id"
        type="wc-admin"
      >
        View dispute
      </Link>,
    ],
    "date": 2020-04-02T02:06:14.000Z,
    "headline": "Payment disputed as Fraudulent.",
    "icon": <t
      className="is-error"
      icon="cross"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents single currency events formats dispute_won events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-04T16:20:50.000Z,
    "headline": "Payment status changed to Disputed: Won.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [
      "Dispute reversal: $100.00",
      "Fee refund: $15.00",
    ],
    "date": 2020-04-04T16:20:50.000Z,
    "headline": <React.Fragment>
      $115.00 USD was added to your 
      <Link
        href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=dummy_po_5eaada696b2d3"
        type="wc-admin"
      >
        Apr 5, 2020 deposit
      </Link>
      .
    </React.Fragment>,
    "icon": <t
      className=""
      icon="plus"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-04T16:20:50.000Z,
    "headline": "Dispute won! The bank ruled in your favor.",
    "icon": <t
      className="is-success"
      icon="notice-outline"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents single currency events formats financing paydown events 1`] = `
Array [
  Object {
    "body": Array [
      <React.Fragment>
        Loan repayment: 
        <Link
          href="/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions&type=charge&filter=advanced&loan_id_is=flxln_1KOKzdR4ByxURRrFX9A65q40"
          type="wc-admin"
        >
          Loan flxln_1KOKzdR4ByxURRrFX9A65q40
        </Link>
      </React.Fragment>,
    ],
    "date": 2022-02-01T12:04:04.000Z,
    "headline": "$110.00 will be subtracted from a future deposit.",
    "icon": <t
      className=""
      icon="minus"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents single currency events formats full_refund events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-04T13:51:06.000Z,
    "headline": "Payment status changed to Refunded.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-04T13:51:06.000Z,
    "headline": "$100.00 USD will be deducted from a future deposit.",
    "icon": <t
      className=""
      icon="minus"
      size={24}
    />,
  },
  Object {
    "body": Array [
      undefined,
    ],
    "date": 2020-04-04T13:51:06.000Z,
    "headline": "A payment of $100.00 USD was successfully refunded.",
    "icon": <t
      className="is-success"
      icon="checkmark"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents single currency events formats partial_refund events 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-03T18:58:01.000Z,
    "headline": "Payment status changed to Partial refund.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-03T18:58:01.000Z,
    "headline": "$50.00 USD will be deducted from a future deposit.",
    "icon": <t
      className=""
      icon="minus"
      size={24}
    />,
  },
  Object {
    "body": Array [
      undefined,
    ],
    "date": 2020-04-03T18:58:01.000Z,
    "headline": "A payment of $50.00 USD was successfully refunded.",
    "icon": <t
      className="is-success"
      icon="checkmark"
      size={24}
    />,
  },
]
`;

exports[`mapTimelineEvents single currency events should not render fee breakup when fee history is not present 1`] = `
Array [
  Object {
    "body": Array [],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": "Payment status changed to Paid.",
    "icon": <t
      className=""
      icon="sync"
      size={24}
    />,
  },
  Object {
    "body": Array [],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": <React.Fragment>
      $59.50 USD was added to your 
      <Link
        href="admin.php?page=wc-admin&path=%2Fpayments%2Fdeposits%2Fdetails&id=dummy_po_5eaada696b281"
        type="wc-admin"
      >
        Apr 2, 2020 deposit
      </Link>
      .
    </React.Fragment>,
    "icon": <t
      className=""
      icon="plus"
      size={24}
    />,
  },
  Object {
    "body": Array [
      undefined,
      "Fee (1.95% + $0.15): $-3.50",
      undefined,
      "Net deposit: $59.50 USD",
    ],
    "date": 2020-04-01T14:37:54.000Z,
    "headline": "A payment of $63.00 USD was successfully charged.",
    "icon": <t
      className="is-success"
      icon="checkmark"
      size={24}
    />,
  },
]
`;
